### PR TITLE
[DOC] Some updates on functions.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by Marenz — APPROVED** [review]

File: docs/contracts/functions.rst [hunk 1]
```diff
@@ -11,8 +11,8 @@ Functions
 Function Parameters and Return Variables
 ========================================
 
-As in JavaScript, functions may take parameters as input. Unlike in JavaScript
-and C, functions may also return an arbitrary number of values as output.
+Functions take typed parameters as input and may, unlike in many other
```
Comment (ChrisChinchilla) [inline docs/contracts/functions.rst:14]: Is "parameters with types" potentially slightly clearer?

File: docs/contracts/functions.rst [hunk 2]
```diff
@@ -107,7 +108,8 @@ Returning Multiple Values
 -------------------------
 
 When a function has multiple return types, the statement ``return (v0, v1, ..., vn)`` can be used to return multiple values.
-The number of components must be the same as the number of return types.
+The number of components must be the same as the number of return variables
+and their types have to match, potentially after an implicit conversion.
```
Comment (ChrisChinchilla) [inline docs/contracts/functions.rst]: I think there's a section on implicit conversion, maybe link to it? Actually I checked, it has no link defined, but I can add to this PR if you think it's a useful idea

Comment (chriseth) [reply docs/contracts/functions.rst]: yes, please do!

File: docs/contracts/functions.rst (hunk 1 above)
Comment (chriseth) [reply docs/contracts/functions.rst:14]: I think I still prefer the current way it is written.

File: docs/contracts/functions.rst [hunk 3]
```diff
@@ -229,7 +231,8 @@ This behaviour is also in line with the ``STATICCALL`` opcode.
 Receive Ether Function
 ======================
 
-A contract can have at most one ``receive`` function, declared using ``receive() external payable``.
+A contract can have at most one ``receive`` function, declared using ``receive() external payable``
+(without the ``function`` keyword).
```
Comment (ChrisChinchilla) [inline docs/contracts/functions.rst:236]: Adding this clarifies in ways, but now I'm confused in other ways, what does the syntax look like? Perhaps an example is useful.

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
I left some small thoughts around clarifications that I am happy to do if you think they are useful

File: docs/contracts/functions.rst (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/contracts/functions.rst]: Done, ended up using an existing link is close enough anyway

**Pull request merged**